wlauto/common/android/workload.py

```diff
@@ -476,6 +481,22 @@ def setup(self, context):
         time.sleep(self.loading_time)
         ReventWorkload.setup(self, context)
 
+        # state detection check if it's enabled in the config
+        if self.check_states:
+            try:
+                self.logger.info("\tChecking workload state...")
+                statedefs_dir = context.resolver.get(File(self, 'state_definitions'))
+                self.device.capture_screen(os.path.join(context.output_directory, "aftersetup.png"))
+                stateCheck = stateDetector.verify_state(os.path.join(context.output_directory,
+                    "aftersetup.png"), statedefs_dir, "setupComplete")
+                if not stateCheck:
+                    raise WorkloadError("Unexpected state after setup")
+            except ResourceError:
```

Comment:
These checks should be happening as part of the init_resources method. That way if the assets cannot be found or are corrupt the workload will not even attempt to run. This will present itself to the end user much sooner than waiting till the workload setup.